tests/src/test/java/io/smallrye/concurrency/test/JTATest.java

```diff
@@ -334,4 +334,40 @@ public void testTransactionWithUT() throws Exception {
             throw e;
         }
     }
+
+    @Test
+    public void testTransactionWithSuspend() throws Exception {
+        ManagedExecutor executor = ManagedExecutor.builder()
+                .maxAsync(2)
+                .propagated(ThreadContext.TRANSACTION)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+
+        try (WeldContainer container = weld.initialize()) {
+            // lookup the transaction manager and the test service
+            UserTransaction ut = container.select(UserTransaction.class).get();
```

Comment:
@mmusgrov I am surprised you don't need Weld JTA artifact here (as in `/tests` module` as well?

Reply:
I agree but I did run all the tests locally with success (I checked out commit ddb2e2c of the upstream MP-C branch - the one before the config stuff was removed).

Reply:
``` bash
mvn test -f tests/pom.xm
[INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 0
```

Reply:
I am debugging where the test is getting UserTransaction from ...

Reply:
So well spotted @manovotn, I had a second CDI producer for UserTransaction in the test module so I removed that.
I also moved the Weld TransactionServices implementation to the `smallrye-concurrency-jta` artefact and created a test jar for it. Both the tck and test modules now depend on this new test jar.

Reply:
Good job!